python/provena-summaries.ipynb

```diff
@@ -20,275 +20,353 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 10,
+   "execution_count": 43,
    "metadata": {},
    "outputs": [],
    "source": [
-    "# This is a small helper class which provides a config object for validation and\n",
-    "# a loader function\n",
-    "import example_workflow_config\n",
+    "from provenaclient import ProvenaClient, Config\n",
+    "from provenaclient.auth import DeviceFlow\n",
+    "from provenaclient.auth.implementations import OfflineFlow\n",
     "\n",
-    "# this contains helpers for interacting with the registry\n",
-    "import registry\n",
-    "\n",
-    "# This is a helper function for managing authentication with Provena\n",
-    "import mdsisclienttools.auth.TokenManager as ProvenaAuth\n",
-    "import mdsisclienttools.datastore.ReadWriteHelper as ProvenaRW\n",
-    "from typing import Dict, Any, Callable\n",
-    "\n",
-    "\n",
-    "import json\n",
-    "import time\n",
-    "import requests"
+    "from pprint import pprint"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 2,
+   "execution_count": 44,
    "metadata": {},
    "outputs": [],
    "source": [
+    "import os\n",
+    "from dotenv import load_dotenv\n",
+    "\n",
     "# Provena config - replace with your Provena instance endpoints\n",
+    "client_config = Config(\n",
+    "    domain=\"dev.rrap-is.com\",\n",
+    "    realm_name=\"rrap\"\n",
+    ")\n",
     "\n",
-    "# Replace the domain with the domain of your Provena instance\n",
-    "PROVENA_DOMAIN = \"dev.rrap-is.com\"\n",
+    "offline_mode = False\n",
     "\n",
-    "# Edit this to point to the Keycloak instance for your Provena instance\n",
-    "kc_endpoint = \"https://auth.dev.rrap-is.com/auth/realms/rrap\"\n",
+    "if offline_mode:\n",
+    "    load_dotenv()\n",
+    "    offline_token=os.getenv('PROVENA_API_TOKEN')\n",
+    "    assert offline_token, \"Offline token must be present in .env file e.g. PROVENA_API_TOKEN=1234.\"\n",
+    "    print(f\"Offline mode activated and token found in .env file.\")\n",
     "\n",
-    "stage = \"DEV\"\n",
-    "registry_endpoint = \"https://registry-api.{}\".format(PROVENA_DOMAIN)\n",
-    "provenance_endpoint = \"https://prov-api.{}\".format(PROVENA_DOMAIN)\n",
-    "data_store_endpoint = \"https://data-api.{}\".format(PROVENA_DOMAIN)\n",
-    "job_endpoint =  \"https://job-api.{}\".format(PROVENA_DOMAIN)"
+    "if not offline_mode:\n",
+    "    auth = DeviceFlow(config=client_config,\n",
+    "                    client_id=\"client-tools\")\n",
+    "else:\n",
+    "    auth = OfflineFlow(config=client_config, offline_token=offline_token)\n",
```

Comment:
The OfflineFlow auth instantiation here is missing positional argument 'client_id' causing cell execution to fail

Reply:
Thanks for picking that up, I have fixed.